ext/pgsql/pgsql.c

```diff
@@ -651,30 +760,29 @@ static void php_pgsql_do_connect(INTERNAL_FUNCTION_PARAMETERS, int persistent)
 			}
 		}
 
-		/* add it to the list */
-		RETVAL_RES(zend_register_resource(pgsql, le_link));
+		object_init_ex(return_value, pgsql_link_ce);
+		link = Z_PGSQL_LINK_P(return_value);
+		link->conn = pgsql;
+		link->hash = zend_string_copy(str.s);
+		link->notices = NULL;
+		link->persistent = 0;
 
 		/* add it to the hash */
-		new_index_ptr.ptr = (void *) Z_RES_P(return_value);
-		new_index_ptr.type = le_index_ptr;
-		zend_hash_update_mem(&EG(regular_list), str.s, (void *) &new_index_ptr, sizeof(zend_resource));
+		ZVAL_COPY(&new_index_ptr, return_value);
```

Comment:
Yeah, you are right, currently, connections are removed from the list whenever they are closed - so in case of objects, then GC should work properly. However, I got a segfault when I changed this to `ZVAL_COPY_VALUE`, so I'm trying to hunt the issue down.